### DISCUSSION TITLE
Version 3.* does not support dotnet 6

### DISCUSSION THREAD

**Discussion by bartlomiejgawel** [JasperFx/marten#2097] [Q&A] [answered]
Is there any particular reason why you require `Microsoft.Extensions.DependencyInjection.Abstractions` up to version 6 exclusively?
https://github.com/JasperFx/marten/blob/3.14/src/Marten/Marten.csproj#L64

I would like to migrate my application to dotnet 6 first and then upgrade the Marten version. Unfortunately, after upgrading the dotnet version, I get the following error:
```
Detected package version outside of dependency constraint: Marten 3.14.2 requires Microsoft.Extensions.DependencyInjection.Abstractions (>= 2.0.0 && < 6.0.0) but version Microsoft.Extensions.DependencyInjection.Abstractions 6.0.0 was resolved.
```

Could you change the above requirement to make it work with dotnet 6?

**Comment by jeremydmiller** [comment, marked as answer]
@bartlomiejgawel 1.) That's a pretty old version of Marten now 2.) Marten v5-alpha has the .Net 6 support. There were other changes and Npgsql issues as well. Moving this to a discussion just to get it out of issues.
